### ISSUE TITLE
add a data structure that specifies a migration: collections to be migrated, source schema and destination schema

### ISSUE THREAD

**Issue by turbomam** [microbiomedata/nmdc-schema#1148]

**Comment by eecavanna** [comment]
Related lines from `nmdc-runtime`:

https://github.com/microbiomedata/nmdc-runtime/blob/7db4d5164c54e0298eba1ee45d5294f0cc278121/demo/metadata_migration/notebooks/migrate_7_8_0_to_8_0_0.ipynb#L403-L408

**Comment by eecavanna** [comment]
I've created a branch for this and will work on it now: 
```shell
git fetch origin
git checkout 1148-add-a-data-structure-that-specifies-a-migration-collections-to-be-migrated-source-schema-and-destination-schema
```

**Comment by eecavanna** [comment]
This feature has been implemented in PR branch #1155 (not merged into `main` yet).
